meshio/permas_io.py

```diff
@@ -132,59 +154,81 @@ def write(
                 (k+1, x[0], x[1], x[2])
                 )
 
-        # Write elements.
-        # translate the number of points per element (aka the type) to a PERMAS
-        # enum.
-        type_to_number = {
-            1: 15,  # point
-            2: 1,  # line
-            3: 2,  # triangle
-            4: 4,  # tetrahedron
-            6: 6,  # prism
-            8: 5,  # hexahedron
+        meshio_to_permas_type = {
+            'vertex': 'NA',
+            'line': 'PLOTL2',
+            'triangle': 'TRIA3',
+            'quad': 'QUAD4',
+            'tetra': 'TET4',
+            'hexahedron': 'HEXE8',
+            'wedge': 'PENTA6',
+            'pyramid': 'PYRA5'
             }
-
-        # PERMAS I/O can only deal with one element type at a time
-        assert len(cells) == 1
-        cell = cells.values()[0]
-
-        for k, c in enumerate(cell):
-            n = len(c)
-            form = '        %8d ' + ' '.join(n * ['%8d']) + '\n'
-            if n == 2:
-                if k == 0:
-                    fh.write('! \n')
-                    fh.write('        $ELEMENT TYPE = PLOTL2 ESET = PLOTL2\n')
-                fh.write(form % (k+1, c[0], c[1]))
-            elif n == 3:
-                if k == 0:
-                    fh.write('! \n')
-                    fh.write('        $ELEMENT TYPE = TRIA3 ESET = TRIA3\n')
-                fh.write(form % (k+1, c[0]+1, c[1]+1, c[2]+1))
-            elif n == 4:
-                if k == 0:
-                    fh.write('!\n')
-                    fh.write('        $ELEMENT TYPE = TET4 ESET = TET4\n')
-                fh.write(form % (k+1, c[0]+1, c[1]+1, c[2]+1, c[3]+1))
-            elif n == 6:
-                if k == 0:
-                    fh.write('!\n')
-                    fh.write('        $ELEMENT TYPE = PENTA6 ESET = PENTA6\n')
-                fh.write(form % (
-                    k+1, c[0]+1, c[1]+1, c[2]+1, c[3]+1, c[4]+1, c[5]+1
-                    ))
-            elif n == 8:
-                if k == 0:
-                    fh.write('!\n')
-                    fh.write('        $ELEMENT TYPE = HEXE8 ESET = HEXE8\n')
-                fh.write(form % (
-                    k+1, c[0]+1, c[1]+1, c[2]+1, c[3]+1,
-                    c[4]+1, c[5]+1, c[6]+1, c[7]+1
-                    ))
+        #
+        # Avoid non-unique element numbers in case of multiple element types by num_ele !!!
+        #
+        num_ele = 0
+        
+        for meshio_type, cell in cells.iteritems():
+            numcells, num_local_nodes = cell.shape
+            permas_type = meshio_to_permas_type[meshio_type]
+            if permas_type != 'NA':
```

Comment:
If you just remove the `if` and the entry from the `dict`, you'll get a `KeyError` if one of the `meshio_type`s is `'vertex'`. That should be fine.